esphome/components/uart/event/__init__.py

```diff
@@ -0,0 +1,90 @@
+import esphome.codegen as cg
+from esphome.components import event, uart
+import esphome.config_validation as cv
+from esphome.const import CONF_EVENT_TYPES, CONF_ID
+from esphome.core import ID
+from esphome.types import ConfigType
+
+from .. import uart_ns
+
+CODEOWNERS = ["@eoasmxd"]
+
+DEPENDENCIES = ["uart"]
+
+UARTEvent = uart_ns.class_("UARTEvent", event.Event, uart.UARTDevice, cg.Component)
+
+
+def validate_event_types(value):
```

Comment:
```suggestion
def validate_event_types(value) -> list[tuple[str, str | list[int]]]:
```

Ideally type value as well